app/metal-controller-manager/controllers/server_controller.go

```diff
@@ -61,24 +68,26 @@ func (r *ServerReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 		return ctrl.Result{RequeueAfter: constants.DefaultRequeueAfter}, err
 	}
 
-	f := func(ready, requeue bool) (ctrl.Result, error) {
-		s.Status.Power = "off"
+	s.Status.Power = "off"
 
-		poweredOn, err := mgmtClient.IsPoweredOn()
-		if err != nil {
-			s.Status.Power = "unknown"
-		}
+	poweredOn, powerErr := mgmtClient.IsPoweredOn()
+	if powerErr != nil {
+		s.Status.Power = "unknown"
+	}
 
-		if poweredOn {
-			s.Status.Power = "on"
-		}
+	if poweredOn {
+		s.Status.Power = "on"
+	}
 
+	f := func(ready, requeue bool) (ctrl.Result, error) {
 		s.Status.Ready = ready
 
 		result := ctrl.Result{Requeue: requeue}
 
-		if err := r.Status().Update(ctx, &s); err != nil {
-			return result, err
+		if err := patchHelper.Patch(ctx, &s, patch.WithOwnedConditions{
+			Conditions: []clusterv1.ConditionType{metalv1alpha1.ConditionPowerCycle},
+		}); err != nil {
+			return result, errors.WithStack(err)
```

Comment:
`errors.WithStack` embeds original stack trace in the log message, not perfect, but better than nothing I guess